python/orion_finance_sdk_py/contracts.py

```
@@ -352,7 +419,15 @@ def create_orion_vault(
         tx_hash = self.w3.eth.send_raw_transaction(signed.raw_transaction)
         tx_hash_hex = tx_hash.hex()
 
-        receipt = self._wait_for_transaction_receipt(tx_hash_hex)
+        try:
+            receipt = self._wait_for_transaction_receipt(tx_hash_hex)
+        except Exception as e:
+            error_str = str(e)
+            if "0xea8e4eb5" in error_str:
+                raise ValueError(
+                    f"Transaction reverted: Manager {account.address} is not whitelisted to create vaults."
+                )
```

Ruff (0.14.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)